readthedocs/core/adapters.py

```diff
@@ -65,3 +77,59 @@ def pre_social_login(self, request, sociallogin):
         sociallogin.email_addresses = [
             email for email in sociallogin.email_addresses if email.primary
         ]
+
+    def _connect_github_app_to_existing_github_account(self, request, sociallogin):
+        """
+        Connect a GitHub App (new integration) account to an existing GitHub account (old integration).
+
+        When a user signs up with the GitHub App we check if there is an existing GitHub account,
+        and if it belongs to the same user, we connect the accounts instead of creating a new one.
+        """
+        provider = sociallogin.account.get_provider()
+
+        # If the provider is not GitHub App, nothing to do.
+        if provider.id != GitHubAppProvider.id:
+            return
+
+        # If the user already signed up with the GitHub App, nothing to do.
+        if sociallogin.is_existing:
+            return
```

Comment:
how do we know that `sociallogin` points to GitHub App here and not to other provider?

Reply:
That is checked above this line.